core/railing/railing.py

```diff
@@ -38,14 +39,9 @@ def make_fill(bm, face, prop):
     bmesh.ops.translate(bm, verts=top_edge.verts, vec=Vector((0., 0., -1.))*prop.corner_post_width/2)
 
     # create railing top
-    ret = bmesh.ops.duplicate(bm, geom=[top_edge])
-    top_dup_edge = filter_geom(ret["geom"], BMEdge)[0]
-    horizon = edge_vector(top_dup_edge).cross(Vector((0., 0., 1.)))
-    up = edge_vector(top_dup_edge)
-    up.rotate(Quaternion(horizon, math.pi/2).to_euler())
-    edge_to_cylinder(bm, top_dup_edge, prop.corner_post_width/2, up)
-    bmesh.ops.translate(bm, verts=top_edge.verts, vec=Vector((0., 0., -1.))*prop.corner_post_width/2)
+    create_railing_top(bm, top_edge, prop)
```

Comment:
Since create_railing_top already has a `map_new_faces` decorator, you need to use the `skip` option so as not to overwrite what it has set. i.e Line 25 should be
```python
@map_new_face(FaceMap.RAILING_POSTS, skip=FaceMap.RAILING_RAILS)
```
of course, this will be the case after you change line 56 as per the comment